identity-issuance/src/main/java/com/android/identity/issuance/funke/FunkeIssuingAuthorityState.kt

```diff
@@ -66,40 +67,83 @@ import kotlin.time.Duration.Companion.seconds
 )
 @CborSerializable
 class FunkeIssuingAuthorityState(
+    // NB: since this object is used to emit notifications, it cannot change, as its state
+    // serves as notification key. So it generally should not have var members, only val.
     val clientId: String,
-    val credentialFormat: CredentialFormat,
-    var issuanceClientId: String = "", // client id in OpenID4VCI protocol
     val credentialIssuerUri: String, // credential offer issuing authority path
+    val credentialConfigurationId: String,
+    val issuanceClientId: String // client id in OpenID4VCI protocol
 ) : AbstractIssuingAuthorityState() {
     companion object {
         private const val TAG = "FunkeIssuingAuthorityState"
 
         private const val DOCUMENT_TABLE = "FunkeIssuerDocument"
 
-        fun getConfiguration(env: FlowEnvironment, credentialFormat: CredentialFormat): IssuingAuthorityConfiguration {
-            val id = when (credentialFormat) {
-                CredentialFormat.SD_JWT_VC -> "funkeSdJwtVc"
-                CredentialFormat.MDOC_MSO -> "funkeMdocMso"
-            }
-            val variant = when (credentialFormat) {
-                CredentialFormat.SD_JWT_VC -> "SD-JWT"
-                CredentialFormat.MDOC_MSO -> "mDoc"
-            }
-            return env.cache(IssuingAuthorityConfiguration::class, id) { configuration, resources ->
-                val logoPath = "funke/logo.png"
-                val logo = resources.getRawResource(logoPath)!!
-                val artPath = "funke/card_art_funke_generic.png"
-                val art = resources.getRawResource(artPath)!!
+        suspend fun getConfiguration(
+            env: FlowEnvironment,
+            issuerUrl: String,
+            credentialConfigurationId: String
+        ): IssuingAuthorityConfiguration {
+            val id = "openid4vci#$issuerUrl#$credentialConfigurationId"
```

Comment:
Per spec they are not allowed to have "#". In this code it also should not be possible for them to have it, but there is a point earlier in the game where we probably should sanitize it to avoid surprises. I'll just add checks in both places, actually.